economy/models.py

```diff
@@ -1,48 +1,114 @@
+import os
+from typing import Union, Optional
+
+from django.conf import settings
 from django.db import models
+from django.db.models import QuerySet, Q
 
 from users.models import User
 
 
 class Transaction(models.Model):
-    sender = models.ForeignKey(User, blank=False, null=False, related_name='sent_transactions')
-    recipient = models.ForeignKey(User, blank=False, null=False, related_name='received_transactions')
+    source = models.ForeignKey('economy.SociBankAccount', related_name='source_transactions', blank=True, null=True,
+                               on_delete=models.SET_NULL)
+    destination = models.ForeignKey('economy.SociBankAccount', related_name='destination_transactions', blank=True,
+                                    null=True, on_delete=models.SET_NULL)
+
     amount = models.IntegerField(blank=False, null=False)
+    product = models.ForeignKey('economy.SociProduct', blank=True, null=True, on_delete=models.SET_NULL)
 
     signed_off_by = models.ForeignKey(User, null=True, related_name='verified_transactions')
     signed_off_time = models.DateTimeField(auto_now_add=True)
 
     @property
-    def valid(self):
+    def is_valid(self):
         return self.signed_off_by is not None
 
-    @property
-    def invalid(self):
-        return not self.valid
-
     def __str__(self):
-        return "Transaction from %s to %s of %d NOK" % (self.sender.name, self.recipient.name, self.amount)
+        return f"Transaction from {self.source.user} to {self.destination.user} of {self.amount} kr"
 
     def __repr__(self):
-        return "Transaction(from=%s, to=%s, amount=%d)" % (self.sender.name, self.recipient.name, self.amount)
+        return f"Transaction(from={self.source.user}, to={self.destination.user}, amount={self.amount})"
+
+    def save(self, force_insert=False, force_update=False, using=None, update_fields=None):
+        if self.source:
```

Comment:
This could potentially be handled in a signal instead of a custom save method.

Reply:
Smart, I agree